client/wsclient_test.go

```diff
@@ -59,6 +72,24 @@ func TestDisconnectWSByServer(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestDisconnectWSByClient(t *testing.T) {
+	// Start a Server.
+	srv := internal.StartMockServer(t)
+
+	defer srv.Close()
+
+	var settings types.StartSettings
+	settings.OpAMPServerURL = "ws://" + srv.Endpoint
+	client := NewWebSocket(testLogger{t})
+	done := make(chan struct{})
+	go func() {
+		startClient(t, settings, client)
+		done <- struct{}{}
+	}()
+	eventually(t, func() bool { <-done; return true })
```

Comment:
I don't think this will provide reliable confirmation that the client connected. `startClient` does not block on connecting to a server, so the `done` channel will likely emit a message almost immediately.